### PR TITLE
windows: update packaging artifacts for MSI and Chocolatey builds

### PR DESCRIPTION
**Pull request by muffins**
## Summary:
The chocolatey validation process checks all of the XML fields for the nuget script. The osquery icon recently moved causing our validation process to fail as the iconUrl field was giving a 404. This updates our flow to the recommended local file for the osquery icon, as per the [`nuspec` docs](https://docs.microsoft.com/en-us/nuget/reference/nuspec#icon). Further, this diff grabs the `tools` and packs directories and drops them where appropriate.

## Test Plan:
* Install osquery from the upstream repo
* Generate a chocolatey package based off of this branch:
```
PS C:\Users\Nicholas\work\repos\osquery-packaging\build> cmake -DOSQUERY_DATA_PATH="C:\" -DCPACK_GENERATOR=NuGet -DOSQUERY_PACKAGE_VERSION="4.8.0" -DOSQUERY_BITNESS=64 ../
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.19042.
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/Nicholas/work/repos/osquery-packaging/build
PS C:\Users\Nicholas\work\repos\osquery-packaging\build> cmake --build . --config Release --target package
Microsoft (R) Build Engine version 16.8.3+39993bd9d for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  CPack: Create package using NuGet
  CPack: Install projects
  CPack: - Install project: osquery [Release]
  CPack: Create package
  Attempting to build package from 'CPack.NuGet.nuspec'.
  Successfully created package 'C:\Users\Nicholas\work\repos\osquery-packaging\build\_CPack_Packages\win64\NuGet\osquery-4.8.0\osquery.4.8.0.nupkg'.
EXEC : warning : NU5110: The script file 'manage-osqueryd.ps1' is outside the 'tools' folder and hence will not be executed during installation of this package. Move it into
the 'tools' folder. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
EXEC : warning : NU5110: The script file 'osquery_utils.ps1' is outside the 'tools' folder and hence will not be executed during installation of this package. Move it into th
e 'tools' folder. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
EXEC : warning : NU5111: The script file 'manage-osqueryd.ps1' is not recognized by NuGet and hence will not be executed during installation of this package. Rename it to ins
tall.ps1, uninstall.ps1 or init.ps1 and place it directly under 'tools'. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
EXEC : warning : NU5111: The script file 'osquery_utils.ps1' is not recognized by NuGet and hence will not be executed during installation of this package. Rename it to insta
ll.ps1, uninstall.ps1 or init.ps1 and place it directly under 'tools'. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
EXEC : warning : NU5111: The script file 'tools\chocolateyBeforeModify.ps1' is not recognized by NuGet and hence will not be executed during installation of this package. Ren
ame it to install.ps1, uninstall.ps1 or init.ps1 and place it directly under 'tools'. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
EXEC : warning : NU5111: The script file 'tools\chocolateyinstall.ps1' is not recognized by NuGet and hence will not be executed during installation of this package. Rename i
t to install.ps1, uninstall.ps1 or init.ps1 and place it directly under 'tools'. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
EXEC : warning : NU5111: The script file 'tools\chocolateyuninstall.ps1' is not recognized by NuGet and hence will not be executed during installation of this package. Rename
 it to install.ps1, uninstall.ps1 or init.ps1 and place it directly under 'tools'. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
EXEC : warning : NU5111: The script file 'tools\osquery_utils.ps1' is not recognized by NuGet and hence will not be executed during installation of this package. Rename it to
 install.ps1, uninstall.ps1 or init.ps1 and place it directly under 'tools'. [C:\Users\Nicholas\work\repos\osquery-packaging\build\package.vcxproj]
  CPack: - package: C:/Users/Nicholas/work/repos/osquery-packaging/build/osquery.4.8.0.nupkg generated.
```
* Run the chocolatey package to ensure that it installs, and that our service comes online as expected.
```
PS C:\Users\Nicholas\Desktop\tmp\testplan> choco install -yf osquery -s . --version 4.8.0 --params='/InstallService'
Chocolatey v0.10.15
Installing the following packages:
osquery
By installing you accept licenses for the packages.
osquery v4.8.0 already installed. Forcing reinstall of version '4.8.0'.
 Please use upgrade if you meant to upgrade to a new version.

osquery v4.8.0 (forced)
osquery package files install completed. Performing other installation steps.
C:\Program Files\osquery\log
True
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 ShimGen has successfully created a shim for osqueryi.exe
 ShimGen has successfully created a shim for osqueryd.exe
 The install of osquery was successful.
  Software install location not explicitly set, could be in package or
  default install location if installer.

Chocolatey installed 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

...

PS C:\Users\Nicholas\Desktop\tmp\testplan> & 'C:\Program Files\osquery\osqueryd\osqueryd.exe' --version
osqueryd.exe version 4.8.0-42-g8900da822-dirty

PS C:\Users\Nicholas\Desktop\tmp\testplan> Get-Service osqueryd

Status   Name               DisplayName
------   ----               -----------
Running  osqueryd           osqueryd
```
* Lastly, run the MSI installer and ensure that it works as expected:
```
PS C:\Users\Nicholas\work\repos\osquery> Get-Service osqueryd

Status   Name               DisplayName
------   ----               -----------
Running  osqueryd           osqueryd

PS C:\Users\Nicholas\work\repos\osquery> & 'C:\Program Files\osquery\osqueryd\osqueryd.exe' --version
osqueryd.exe version 4.8.0-42-g8900da822-dirty

osquery> select name,version,uninstall_string from programs where name like '%osquery%';
+---------+---------+------------------------------------------------------+
| name    | version | uninstall_string                                     |
+---------+---------+------------------------------------------------------+
| osquery | 4.8.0   | MsiExec.exe /I{63C34DB3-9D28-445F-875A-E27ADA8AEECC} |
+---------+---------+------------------------------------------------------+
```

### CODE REVIEW

**Comment by muffins** [comment]
I need to get a small cmake change shipped to the upstream osquery before this can ship, as we're also going to try and address some deprecated nuget spec issues.

**Review by directionless — DISMISSED** [review]

**Comment by muffins** [comment]
I have another change coming that brings the osquery query packs as well, just waiting for that build so I can't test generating the packages and I'll update this PR.

Further we are still waiting for [this PR](https://github.com/osquery/osquery/pull/7148) to land before we can run these changes off of master

**Comment by muffins** [comment]
This should be g2g, @directionless or @theopolis mind giving this a look when you get a chance?

File: platform/windows/common.cmake [hunk 1]
```diff
@@ -26,22 +27,22 @@ set(file_name_list
 
 foreach(directory_name ${directory_name_list})
   install(
-    DIRECTORY "${OSQUERY_DATA_PATH}/Program Files (x86)/osquery/${directory_name}"
+    DIRECTORY "${OSQUERY_DATA_PATH}/Program Files/osquery/${directory_name}"
```
Comment (muffins) [inline platform/windows/common.cmake:29]: @alessandrogario or @directionless do you know if us migrating to the non-`(x86)` Program Files is going to cause any issues? I feel like we should default to `Program Files` as we typically generate the x64 based binaries yes?

Comment (directionless) [reply platform/windows/common.cmake:29]: The internet says:
> Starting with Windows Vista, the 64-bit versions of Windows have two Program Files folders. The regular Program Files folder holds 64-bit applications, while "Program Files (x86)" is used for 32-bit applications.

So I think I think it is correct to drop the `(x86)`

**Review by directionless — DISMISSED** [review]
This seems fine to me. As mentioned in the other PR I feel weird about `packs` but....

File: platform/windows/common.cmake (hunk 1 above)
Comment (muffins) [reply platform/windows/common.cmake:29]: :) Never did trust that "internet"

**Comment by muffins** [comment]
Removed the packs reference, @directionless or @theopolis one more time? Apologies for the noise!

File: platform/windows/nuget.cmake [hunk 2]
```diff
@@ -21,13 +33,14 @@ set(CPACK_NUGET_PACKAGE_DESCRIPTION "
   `--params`. For example: `--params='/InstallService'`.
   "
 )
+set(OSQUERY_RAW "https://raw.githubusercontent.com/osquery/osquery/")
```
Comment (theopolis) [inline platform/windows/nuget.cmake]: Is this used?

File: platform/windows/nuget.cmake [hunk 3]
```diff
@@ -21,13 +33,14 @@ set(CPACK_NUGET_PACKAGE_DESCRIPTION "
...
 set(OSQUERY_REPO "https://github.com/osquery/osquery/")
```
Comment (theopolis) [inline platform/windows/nuget.cmake]: Is this no longer used?

**Review by theopolis — CHANGES_REQUESTED** [review]

File: platform/windows/nuget.cmake (hunk 3 above)
Comment (muffins) [reply platform/windows/nuget.cmake]: Still used for the release notes reference.

**Review by theopolis — APPROVED** [review]

**Pull request merged**